diff --git a/unsloth_zoo/llama_cpp.py b/unsloth_zoo/llama_cpp.py
--- a/unsloth_zoo/llama_cpp.py
+++ b/unsloth_zoo/llama_cpp.py
@@ -32,8 +32,7 @@
 import contextlib
 import os
 
-LLAMA_CPP_CONVERT_FILE = \
-    "https://github.com/ggerganov/llama.cpp/raw/refs/heads/master/convert_hf_to_gguf.py"
+LLAMA_CPP_CONVERT_FILE = "llama.cpp/unsloth_convert_hf_to_gguf.py"
 
 COMMANDS_NOT_FOUND = (
     "command not found",
@@ -348,11 +347,12 @@
     # All Unsloth Zoo code licensed under LGPLv3
     # Downloads from llama.cpp's Github repo
     try:
-        converter_latest = requests.get(LLAMA_CPP_CONVERT_FILE).content
+        with open(LLAMA_CPP_CONVERT_FILE, 'rb') as f:
+            converter_latest = f.read()
     except:
         raise RuntimeError(
             f"Unsloth: Could not obtain `{LLAMA_CPP_CONVERT_FILE}`.\n"\
-            f"Maybe you don't have internet ocnnection?"
+            f"Is llama.cpp installed?"
         )
 
     # Get all supported models